packages/backend/src/apps/postman/common/send-blacklist-email.ts

```diff
@@ -0,0 +1,146 @@
+import { DateTime } from 'luxon'
+
+import { createRedisClient, REDIS_DB_INDEX } from '@/config/redis'
+import { sendEmail } from '@/helpers/send-email'
+
+const MAX_LENGTH = 80
+const BLACKLIST_REQUEST_FORM_URL =
+  'https://form.gov.sg/6661900c502545e6788fa3c4'
+const redisClient = createRedisClient(REDIS_DB_INDEX.PIPE_ERRORS)
+
+interface SendBlacklistEmailProps {
+  flowName: string
+  flowId: string
+  userEmail: string
+  executionId: string
+  blacklistedRecipients: string[]
+}
+
+interface BlacklistEmailProps {
+  flowName: string
+  userEmail: string
+  executionId: string
+  blacklistedRecipients: string[]
+}
+
+type BlacklistEmailFormProps = Omit<BlacklistEmailProps, 'flowName'>
+
+const blacklistRedisKey = (flowId: string, email: string) =>
+  `blacklist-alert:${flowId}:${email}`
+
+function truncateFlowName(flowName: string) {
+  return flowName.length > MAX_LENGTH
+    ? flowName.slice(0, MAX_LENGTH) + '...'
+    : flowName
+}
+
+export function createRequestBlacklistFormLink({
+  userEmail,
+  executionId,
+  blacklistedRecipients,
+}: BlacklistEmailFormProps) {
+  const queryParams = new URLSearchParams({
+    '666190bbfdd7e4abe8973b76': blacklistedRecipients.join(','),
```

Comment:
comment: we also have [`urlcat`](https://www.npmjs.com/package/urlcat) for this sort of work